src/ibonwe/shape/diamond.cpp

```diff
@@ -0,0 +1,53 @@
+#include <iostream>
+
+#include "ibonwe/shape/diamond.h"
+
+namespace ibonwe {
+namespace shape {
+
+/**
+ * The default diamond constructor
+ */
+ibonwe::shape::diamond::diamond()
+{
+    character = ' ';
+    width     = 0;
+    height    = 0;
+}
+
+/**
+ * The parameterised diamond constructor
+ *
+ * @param char aCharacter A diamond character
+ * @param int  aHeight    A diamond height
+ */
+ibonwe::shape::diamond::diamond(const char aCharacter, const int aHeight)
+{
+    character = aCharacter;
+    width     = aHeight;
+    height    = aHeight;
+}
+
+/**
+ * The purpose of this method is to draw a diamond
+ */
+void ibonwe::shape::diamond::draw()
```

Comment:
I am wondering if this can't be done more elegantly. This code is a little complicated and also is not necessarily going to perform really fast either for big numbers of height.